pkg/kudoctl/packages/types.go

```diff
@@ -27,25 +28,59 @@ type Resources struct {
 	Instance        *v1beta1.Instance
 }
 
-type Parameter []v1beta1.Parameter
+// Modified v1beta1.Parameter that allows for defaults provided as YAML.
+type Parameter struct {
+	DisplayName string                `json:"displayName,omitempty"`
+	Name        string                `json:"name,omitempty"`
+	Description string                `json:"description,omitempty"`
+	Required    *bool                 `json:"required,omitempty"`
+	Default     interface{}           `json:"default,omitempty"`
+	Trigger     string                `json:"trigger,omitempty"`
+	Type        v1beta1.ParameterType `json:"type,omitempty"`
+}
 
-// Templates is a map of file names and stringified files in the template folder of an operator
-type Templates map[string]string
+type Parameters []Parameter
 
 // Len returns the number of params.
 // This is needed to allow sorting of params.
-func (p Parameter) Len() int { return len(p) }
+func (p Parameters) Len() int { return len(p) }
 
 // Swap swaps the position of two items in the params slice.
 // This is needed to allow sorting of params.
-func (p Parameter) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+func (p Parameters) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
 
 // Less returns true if the name of a param a is less than the name of param b.
 // This is needed to allow sorting of params.
-func (p Parameter) Less(x, y int) bool {
+func (p Parameters) Less(x, y int) bool {
 	return p[x].Name < p[y].Name
 }
 
+func (p Parameters) ToAPI() ([]v1beta1.Parameter, error) {
```

Comment:
could we godoc this please?

Reply:
I'm surprised this passed our linters... 🤔

Reply:
`golangci-lint` ignores these `golint` documentation errors :(
We have 75 places where exported functions aren't documented correctly.
```
$ golint ./pkg/... | grep exported | wc -l
      75
```

Reply:
I created https://github.com/kudobuilder/kudo/issues/1392 to track that `golangci-lint` by default excluded some linter output, including these doc strings.